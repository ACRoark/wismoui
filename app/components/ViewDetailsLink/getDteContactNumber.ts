const getDteContactNumber = (category: string): string => {
  switch (category) {
    case 'R975':
      return '3132356655';
    case 'R980':
      return '8004416698';
    case 'R981':
      return '8558517152';
    case 'R982':
      return '8772000438';
    case 'R983':
      return '8663112244';
    case 'R984':
      return '8004774747';
    case 'R987':
      return '8558387258';
    default:
      console.error('Invalid Category');
      return '';
  }
};

export default getDteContactNumber;
