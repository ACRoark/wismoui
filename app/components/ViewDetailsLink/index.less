@import (reference) '../../styles/themes/index.less';

.@{dte-prefix}-view-details-modal-content {
  display: block;
  text-align: left;

  .@{dte-prefix}-view-details-status-history-title {
    color: @dte-gray-700;
    display: block;
    font-weight: @dte-wismo-font-weight-semi-bold;
    padding: 20px 0;
    text-align: left;
  }

  .website {
    font-weight: @dte-wismo-font-weight-semi-bold;
    text-decoration: underline;
  }
}
