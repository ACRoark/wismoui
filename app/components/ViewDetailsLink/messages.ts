import { defineMessages } from 'react-intl';

export const scope = 'wismo.components.view-details-link';

export default defineMessages({
  AD_title: {
    id: `${scope}.AD-title`,
  },
  AD_R975_description: {
    id: `${scope}.AD-R975-description`,
  },
  AD_R980_description: {
    id: `${scope}.AD-R980-description`,
  },
  AD_R981_description: {
    id: `${scope}.AD-R981-description`,
  },
  AD_R982_description: {
    id: `${scope}.AD-R982-description`,
  },
  AD_R983_description: {
    id: `${scope}.AD-R983-description`,
  },
  AN_R975_description: {
    id: `${scope}.AN-R975-description`,
  },
  AN_R980_description: {
    id: `${scope}.AN-R980-description`,
  },
  AN_R981_description: {
    id: `${scope}.AN-R981-description`,
  },
  AN_R982_description: {
    id: `${scope}.AN-R982-description`,
  },
  AN_R983_description: {
    id: `${scope}.AN-R983-description`,
  },
  AN_title: {
    id: `${scope}.AN-title`,
  },
  AP_R975_description: {
    id: `${scope}.AP-R975-description`,
  },
  AP_R980_description: {
    id: `${scope}.AP-R980-description`,
  },
  AP_R981_description: {
    id: `${scope}.AP-R981-description`,
  },
  AP_R982_description: {
    id: `${scope}.AP-R982-description`,
  },
  AP_R983_description: {
    id: `${scope}.AP-R983-description`,
  },
  AP_title: {
    id: `${scope}.AP-title`,
  },
  CANCELED_CGI_description : {
    id: `${scope}.CANCELED-CGI-description`,
  },
  CANCELED_CGI_title : {
    id: `${scope}.CANCELED-CGI-title`,
  },
  CANCELED_description : {
    id: `${scope}.CANCELED-description`,
  },
  CANCELED_INC_description : {
    id: `${scope}.CANCELED-INC-description`,
  },
  CANCELED_INC_title : {
    id: `${scope}.CANCELED-INC-title`,
  },
  CANCELED_title : {
    id: `${scope}.CANCELED-title`,
  },
  CL_R975_description: {
    id: `${scope}.CL-R975-description`,
  },
  CL_R980_description: {
    id: `${scope}.CL-R980-description`,
  },
  CL_R981_description: {
    id: `${scope}.CL-R981-description`,
  },
  CL_R982_description: {
    id: `${scope}.CL-R982-description`,
  },
  CL_R983_description: {
    id: `${scope}.CL-R983-description`,
  },
  CL_title: {
    id: `${scope}.CL-title`,
  },
  customerVerification: {
    id: `${scope}.customer-verification`,
  },
  DELAYED_description: {
    id: `${scope}.delayed-description`,
  },
  DELAYED_title: {
    id: `${scope}.delayed-title`,
  },
  DR_R975_description: {
    id: `${scope}.DR-R975-description`,
  },
  DR_R980_description: {
    id: `${scope}.DR-R980-description`,
  },
  DR_R981_description: {
    id: `${scope}.DR-R981-description`,
  },
  DR_R982_description: {
    id: `${scope}.DR-R982-description`,
  },
  DR_R983_description: {
    id: `${scope}.DR-R983-description`,
  },
  DR_title: {
    id: `${scope}.DR-title`,
  },
  ES_R975_description: {
    id: `${scope}.ES-R975-description`,
  },
  ES_R980_description: {
    id: `${scope}.ES-R980-description`,
  },
  ES_R981_description: {
    id: `${scope}.ES-R981-description`,
  },
  ES_R982_description: {
    id: `${scope}.ES-R982-description`,
  },
  ES_R983_description: {
    id: `${scope}.ES-R983-description`,
  },
  ES_R986_description: {
    id: `${scope}.ES-R986-description`,
  },
  ES_R987_description: {
    id: `${scope}.ES-R987-description`,
  },
  ES_title: {
    id: `${scope}.ES-title`,
  },
  FV_R975_description: {
    id: `${scope}.FV-R975-description`,
  },
  FV_R980_description: {
    id: `${scope}.FV-R980-description`,
  },
  FV_R981_description: {
    id: `${scope}.FV-R981-description`,
  },
  FV_R982_description: {
    id: `${scope}.FV-R982-description`,
  },
  FV_R983_description: {
    id: `${scope}.FV-R983-description`,
  },
  FV_title: {
    id: `${scope}.FV-title`,
  },
  FW_R975_description: {
    id: `${scope}.FW-R975-description`,
  },
  FW_R980_description: {
    id: `${scope}.FW-R980-description`,
  },
  FW_R981_description: {
    id: `${scope}.FW-R981-description`,
  },
  FW_R982_description: {
    id: `${scope}.FW-R982-description`,
  },
  FW_R983_description: {
    id: `${scope}.FW-R983-description`,
  },
  FW_title: {
    id: `${scope}.FW-title`,
  },
  LD_R983_description: {
    id: `${scope}.LD-R983-description`,
  },
  LD_title: {
    id: `${scope}.LD-title`,
  },
  PA_R984_description: {
    id: `${scope}.PA-R984-description`,
  },
  PA_R985_description: {
    id: `${scope}.PA-R985-description`,
  },
  PA_title: {
    id: `${scope}.PA-title`,
  },
  PM_R984_description: {
    id: `${scope}.PM-R984-description`,
  },
  PM_R985_description: {
    id: `${scope}.PM-R985-description`,
  },
  PM_title: {
    id: `${scope}.PM-title`,
  },
  PR_R975_description: {
    id: `${scope}.PR-R975-description`,
  },
  PR_R980_description: {
    id: `${scope}.PR-R980-description`,
  },
  PR_R981_description: {
    id: `${scope}.PR-R981-description`,
  },
  PR_R982_description: {
    id: `${scope}.PR-R982-description`,
  },
  PR_R983_description: {
    id: `${scope}.PR-R983-description`,
  },
  PR_title: {
    id: `${scope}.PR-title`,
  },
  RE_R975_description: {
    id: `${scope}.RE-R975-description`,
  },
  RE_R980_description: {
    id: `${scope}.RE-R980-description`,
  },
  RE_R981_description: {
    id: `${scope}.RE-R981-description`,
  },
  RE_R982_description: {
    id: `${scope}.RE-R982-description`,
  },
  RE_R983_description: {
    id: `${scope}.RE-R983-description`,
  },
  RE_title: {
    id: `${scope}.RE-title`,
  },
  RF_R975_description: {
    id: `${scope}.RF-R975-description`,
  },
  RF_R980_description: {
    id: `${scope}.RF-R980-description`,
  },
  RF_R981_description: {
    id: `${scope}.RF-R981-description`,
  },
  RF_R982_description: {
    id: `${scope}.RF-R982-description`,
  },
  RF_R983_description: {
    id: `${scope}.RF-R983-description`,
  },
  RF_R986_description: {
    id: `${scope}.RF-R986-description`,
  },
  RF_R987_description: {
    id: `${scope}.RF-R987-description`,
  },
  RF_title: {
    id: `${scope}.RF-title`,
  },
  SC_R975_description: {
    id: `${scope}.SC-R975-description`,
  },
  SC_R980_description: {
    id: `${scope}.SC-R980-description`,
  },
  SC_R981_description: {
    id: `${scope}.SC-R981-description`,
  },
  SC_R982_description: {
    id: `${scope}.SC-R982-description`,
  },
  SC_R983_description: {
    id: `${scope}.SC-R983-description`,
  },
  SC_title: {
    id: `${scope}.SC-title`,
  },
  status_history_title: {
    id: `${scope}.status-history-title`,
  },
  title: {
    id: `${scope}.title`,
  },
  website: {
    id: `${scope}.website`,
  },
  WP_R984_description: {
    id: `${scope}.WP-R984-description`,
  },
  WP_R985_description: {
    id: `${scope}.WP-R985-description`,
  },
  WP_title: {
    id: `${scope}.WP-title`,
  },
});
