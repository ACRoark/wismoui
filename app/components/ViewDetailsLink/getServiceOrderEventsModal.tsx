import React from 'react';
import { FormattedMessage } from 'react-intl';

import DteContactNumber from 'components/DteContactNumber';
import useConfig from 'hooks/useConfig';
import IModalConfig from 'types/IModalConfig';
import { ServiceOrderEventStatus } from 'types/ServiceOrderEventStatus';

import './index.less';
import messages from './messages';

export const getServiceOrderEventsModal = (status: ServiceOrderEventStatus): IModalConfig => {
  const { urls } = useConfig();
  const modalTitle = <FormattedMessage {...messages[`${status}_title`]} />;
  const modalContent = (
    <span className="dte-wismo-view-details-modal-content">
      <FormattedMessage
        {...messages[`${status}_description`]}
        values={{
          phoneNumber: <DteContactNumber />,
          website: (
            <a className="website" href={urls.backToMIMO}>
              <FormattedMessage {...messages.website} />
            </a>
          ),
        }}
      />
    </span>
  );

  return {
    content: modalContent,
    title: modalTitle,
  };
};

export default getServiceOrderEventsModal;
