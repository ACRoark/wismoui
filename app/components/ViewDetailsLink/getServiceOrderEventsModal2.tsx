import React from 'react';
import { FormattedMessage } from 'react-intl';

import DteContactNumber from 'components/DteContactNumber';
import IModalConfig from 'types/IModalConfig';
import { ServiceOrderEventStatus } from 'types/ServiceOrderEventStatus';

import './index.less';
import messages from './messages';

const getServiceOrderEventsModal2 = (status: ServiceOrderEventStatus, url: string): IModalConfig => {
  const modalTitle = <FormattedMessage {...messages[`${status}_title`]} />;
  const modalContent = (
    <span className="dte-wismo-view-details-modal-content">
      <FormattedMessage
        {...messages[`${status}_description`]}
        values={{
          phoneNumber: <DteContactNumber />,
          website: (
            <a className="website" href={url}>
              <FormattedMessage {...messages.website} />
            </a>
          ),
        }}
      />
    </span>
  );

  return {
    content: modalContent,
    title: modalTitle,
  };
};

export default getServiceOrderEventsModal2;
