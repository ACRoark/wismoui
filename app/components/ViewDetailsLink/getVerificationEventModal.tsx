import * as _ from 'lodash';
import React from 'react';
import { FormattedMessage } from 'react-intl';

import IModalConfig from 'types/IModalConfig';
import IVerificationEvent from 'types/IVerificationEvent';

import './index.less';
import messages from './messages';
import ModalContent from './ModalContent';

const getVerificationEventModal = (
  event: IVerificationEvent,
  events: IVerificationEvent[],
  orderCreatedDate: string,
): IModalConfig => {
  const code = _.last(event?.verificationEventStatusUpdates).verificationCode;

  return {
    className: 'dte-wismo-modal',
    content: <ModalContent event={event} events={events} orderCreatedDate={orderCreatedDate} />,
    title: <FormattedMessage {...messages[`${code}_title`]} />,
  };
};

export default getVerificationEventModal;
