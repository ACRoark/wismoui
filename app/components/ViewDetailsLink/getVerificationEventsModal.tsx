import * as _ from 'lodash';
import React, { useEffect, useState } from 'react';
import { FormattedMessage } from 'react-intl';

import DateStamp from 'components/DateStamp';
import DteContactNumber from 'components/DteContactNumber';
import VerificationHistory from 'components/Timelines/VerificationHistory';
import useConfig from 'hooks/useConfig';
import encryptionApiClient from 'services/encryptionApiClient';
import IModalConfig from 'types/IModalConfig';
import IVerificationEvent from 'types/IVerificationEvent';

import getDteContactNumber from './getDteContactNumber';
import './index.less';
import messages from './messages';

const getVerificationEventsModal = (events: IVerificationEvent[], orderCreatedDate: string): IModalConfig => {
  let category;
  let code;
  const [encryptedVerificationEventId, setEncryptedVerificationEventId] = useState('');
  const formattedOrderCreatedDate = orderCreatedDate && <DateStamp value={orderCreatedDate} />;
  let modalContent;
  let modalTitle;
  let showVerificationHistory;
  let verificationEventId;

  if (events && events.length) {
    const lastVerificationEvent = _.last(events);
    category = lastVerificationEvent.verificationCategory;
    code = _.last(lastVerificationEvent.verificationEventStatusUpdates).verificationCode;
    modalTitle = <FormattedMessage {...messages[`${code}_title`]} />;
    verificationEventId = lastVerificationEvent.verificationEventId;

    const contactNumber = getDteContactNumber(category);
    showVerificationHistory = !['AP', 'PM', 'WP'].includes(code);
    const { urls } = useConfig();

    // append the tokenized verification event ID to the url.documentSubmission here
    useEffect((): void => {
      if (verificationEventId) {
        const getEncryptedVerificationEventId = async (): Promise<void> => {
          const encryptedTokenValue = await encryptionApiClient.encryptTokenAsync(
            urls.encryptTokenApi,
            verificationEventId,
          );

          setEncryptedVerificationEventId(encryptedTokenValue);
        };

        getEncryptedVerificationEventId();
      }
    }, []);

    const appendedDocumentSubmissionUrl = `${urls.documentSubmission}${encryptedVerificationEventId}`;

    modalContent = (
      <span className="dte-wismo-view-details-modal-content">
        <FormattedMessage
          {...messages[`${code}_${category}_description`]}
          values={{
            a: (chunks: React.ReactNode): React.ReactNode => (
              <a className="website" href={appendedDocumentSubmissionUrl} target="_blank">
                {chunks}
              </a>
            ),
            assistance: (...msg: React.ReactNode[]): React.ReactNode => (
              <a className="website" href={urls.assistanceURL} target="_blank">
                {msg}
              </a>
            ),
            li: (...msg: React.ReactNode[]): React.ReactNode => <li>{msg}</li>,
            login: (...msg: React.ReactNode[]): React.ReactNode => (
              <a className="website" href={urls.signIn} target="_blank">
                {msg}
              </a>
            ),
            p: (...msg: React.ReactNode[]): React.ReactNode => <p>{msg}</p>,
            ol: (...msg: React.ReactNode[]): React.ReactNode => <ol>{msg}</ol>,
            mimo: (...msg: React.ReactNode[]): React.ReactNode => (
              <a className="website" href={urls.backToMIMO} target="_blank">
                {msg}
              </a>
            ),
            orderCreatedDate: formattedOrderCreatedDate,
            ul: (...msg: React.ReactNode[]): React.ReactNode => <ul>{msg}</ul>,
            phoneNumber: <DteContactNumber value={contactNumber} />,
          }}
        />
        {showVerificationHistory && (
          <>
            <span className="dte-wismo-view-details-status-history-title">
              <FormattedMessage {...messages.status_history_title} />
            </span>
            <VerificationHistory verificationEvents={events} />
          </>
        )}
      </span>
    );
  }

  return {
    className: 'dte-wismo-modal',
    content: modalContent,
    title: modalTitle,
  };
};

export default getVerificationEventsModal;
