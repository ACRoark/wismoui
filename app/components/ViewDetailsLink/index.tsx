import * as _ from 'lodash';
import React from 'react';
import { FormattedMessage } from 'react-intl';

import Button from 'components/Button';
import useModal from 'hooks/useModal';
import useThunkDispatch from 'hooks/useThunkDispatch';
import { trackButtonClick } from 'store/actions';
import IModalConfig from 'types/IModalConfig';
import IOrderRequest from 'types/IOrderRequest';
import getServiceDate from 'utils/getServiceDate';

import getServiceOrderEventsModal from './getServiceOrderEventsModal';
import getVerificationEventsModal from './getVerificationEventsModal';

import './index.less';
import messages from './messages';

interface IViewDetailsLinkProps {
  className?: string;
  orderCreatedDate: string;
  serviceRequest: IOrderRequest;
}

const ViewDetailsLink: React.FC<IViewDetailsLinkProps> = (props: IViewDetailsLinkProps): React.ReactElement => {
  const { className, orderCreatedDate, serviceRequest } = props;
  const { serviceOrderEvents, verificationEvents } = serviceRequest;

  let serviceOrderEventStatus;

  if (serviceOrderEvents && serviceOrderEvents.length) {
    if (_.last(serviceOrderEvents).latestStatus) {
      serviceOrderEventStatus = _.last(serviceOrderEvents).latestStatus.serviceOrderEventStatus;
    }
  }

  const dispatch = useThunkDispatch();
  const { showModal } = useModal();
  const serviceDate = getServiceDate(serviceRequest) || orderCreatedDate;
  const isVerificationComplete = serviceOrderEvents && serviceOrderEvents.length;

  const emptyModal = { content: null, title: null };

  const ServiceOrderEventsModal = serviceOrderEventStatus
    ? getServiceOrderEventsModal(serviceOrderEventStatus)
    : emptyModal;

  const VerificationEventsModal =
    verificationEvents && verificationEvents.length
      ? getVerificationEventsModal(verificationEvents, serviceDate)
      : emptyModal;

  const modal: IModalConfig = isVerificationComplete ? ServiceOrderEventsModal : VerificationEventsModal;

  const trackButtonAndOpenModal = (): void => {
    dispatch(trackButtonClick('view details link'));
    showModal(modal);
  };

  return (
    <Button className={className} onClick={trackButtonAndOpenModal} type="link">
      <FormattedMessage {...messages.title} />
    </Button>
  );
};

export default ViewDetailsLink;
