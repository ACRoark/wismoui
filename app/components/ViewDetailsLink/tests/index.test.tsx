import React from 'react';
import { Provider } from 'react-redux';

import { testConfig } from 'components/constants';
import createFakeOrder from 'components/createFakeOrder';
import useConfig from 'hooks/useConfig';
import useModal from 'hooks/useModal';
import { ModalProvider } from 'providers/ModalProvider';
import encryptionApiClient from 'services/encryptionApiClient';
import { createMockStore, createSnapshotWithIntl, renderWithIntl } from 'testing/utils';
import IModalConfig from 'types/IModalConfig';
import IServiceOrderEvent from 'types/IServiceOrderEvent';
import IVerificationEvent from 'types/IVerificationEvent';

import ViewDetailsLink from '../index';

import { SUPPORTED_LOCALES } from 'locales';
import { FormattedMessage } from 'react-intl';
import messages from '../messages';

jest.mock('hooks/useConfig');
jest.mock('hooks/useModal');
jest.mock('services/encryptionApiClient');

const fakeServiceRequest = createFakeOrder('MI11223344').orderRequests[0];
const requestedTestDate = '2020-05-28';
const verificationCode = 'AD';
const verificationCategory = 'R975';
const verificationEventsActiveBpem: IVerificationEvent[] = [{
    verificationCategory: 'R975',
    verificationEventId: '123123',
    verificationEventStatusUpdates: [
      {
        verificationCode: 'AD',
        verificationEventStatus: 'ACTIVE',
        createdAtDate: '20200520',
        createdAtTime: '092215',
      },
      ],
}];
const mockConfig = useConfig as jest.MockedFunction<typeof useConfig>;
const mockEncryption = encryptionApiClient.encryptTokenAsync as jest.MockedFunction<typeof encryptionApiClient.encryptTokenAsync>;
const mockModal = useModal as jest.MockedFunction<typeof useModal>;

const modalTitle = <FormattedMessage {...messages.customerVerification} />;
const store = createMockStore();

const modalContentForVerification = (
  <span className="view-details-modal-content">
        <FormattedMessage
          {...messages[`${verificationCode}_${verificationCategory}_description`]}
          values={{
            a: (chunks: React.ReactNode): React.ReactNode => (
              <a href={'http://www.senddocuments.com'}
                 target={'_blank'}>{chunks}</a>
            ),
            li: (msg: React.ReactNode): React.ReactNode => (
              <li>
                {msg}
              </li>
            ),
            p: (...msg: React.ReactNode[]): React.ReactNode => (
              <p>
                {msg}
              </p>
            ),
            ol: (...msg: React.ReactNode[]): React.ReactNode => (
              <ol>
                {msg}
              </ol>
            ),
            orderCreatedDate: requestedTestDate,
            ul: (...msg: React.ReactNode[]): React.ReactNode => (
              <ul>
                {msg}
              </ul>
            ),
          }}
        />
  </span>
);

const modalContentForServiceOrderEvents = (
  <span className="dte-wismo-view-details-modal-content">
      <FormattedMessage
        {...messages[`${status}_description`]}
        values={{
          phoneNumber: (
            <a className="phone-number" href="tel:+8004774747">
              800.477.4747
            </a>
          ),
          website: (
            <a href={'http://www.doo-dee-doo.com'}>
              <FormattedMessage {...messages.website} />
            </a>
          ),
        }}
      />
    </span>
);
const modalConfigForVerification: IModalConfig = {
  content: modalContentForVerification,
  title: modalTitle,
};

const modalConfigForServiceOrderEvents: IModalConfig = {
  content: modalContentForServiceOrderEvents,
  title: modalTitle,
};

const serviceOrderEventsCanceledStatus: IServiceOrderEvent[] = [
  {
    appointment: {
      date: '2020-05-22',
      slotType: 'AM',
    },
    latestStatus: {
      createdAtDate: '20200522',
      createdAtTime: '223443',
      serviceOrderEventStatus: 'CANCELED',
    },
    product: 'EFR_D1',
    serviceOrderCategory: null,
    serviceOrderId: 'abc123',
  },
];

const expectComponentToMatchSnapshotForVerification = (language: string): void => {
  mockConfig.mockReturnValue({ ...testConfig });
  mockEncryption.mockReturnValue(new Promise((): string => ''));
  mockModal.mockReturnValue({
    hideModal: jest.fn(),
    showModal: jest.fn(),
  });

  const serviceRequest = {
    ...fakeServiceRequest,
    verificationEvents: verificationEventsActiveBpem,
  };

  const tree = createSnapshotWithIntl(
    <Provider store={store}>
      <ModalProvider {...modalConfigForVerification}>
        <ViewDetailsLink orderCreatedDate={requestedTestDate}
                         serviceRequest={serviceRequest}
        />
      </ModalProvider>
    </Provider>, language);

  expect(tree).toMatchSnapshot();
};

const expectComponentToMatchSnapshotForServiceOrderEvents = (language: string): void => {
  mockConfig.mockReturnValue({ ...testConfig });
  mockEncryption.mockReturnValue(new Promise((): string => ''));
  mockModal.mockReturnValue({
    hideModal: jest.fn(),
    showModal: jest.fn(),
  });

  const serviceRequest = {
    ...fakeServiceRequest,
    serviceOrderEvents: serviceOrderEventsCanceledStatus,
  };

  const tree = createSnapshotWithIntl(
    <Provider store={store}>
      <ModalProvider {...modalConfigForServiceOrderEvents}>
        <ViewDetailsLink
          orderCreatedDate="2020-07-31"
          serviceRequest={serviceRequest}
        />
      </ModalProvider>
    </Provider>, language);

  expect(tree).toMatchSnapshot();
};

const shouldNotLogErrors = (language: string): void => {
  mockConfig.mockReturnValue({ ...testConfig });
  mockEncryption.mockReturnValue(new Promise((): string => ''));
  mockModal.mockReturnValue({
    hideModal: jest.fn(),
    showModal: jest.fn(),
  });

  const spy = jest.spyOn(global.console, 'error');
  const serviceRequest = {
    ...fakeServiceRequest,
    verificationEvents: verificationEventsActiveBpem,
  };

  renderWithIntl(
    <Provider store={store}>
      <ModalProvider {...modalConfigForVerification}>
        <ViewDetailsLink orderCreatedDate="2020-05-28"
                       serviceRequest={serviceRequest}
        />
      </ModalProvider>
    </Provider>, language);

  expect(spy).not.toHaveBeenCalled();
};

describe('<ViewDetailsLink />', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    it(`should not log errors in console when the valid content and title values are provided (${language})`, (): void =>
      shouldNotLogErrors(language));

    it(`should render the link when a valid verificationCategory, verificationCode, and wantDate are provided (${language})`, (): void =>
      expectComponentToMatchSnapshotForVerification(language));

    it(`should render the link when a valid serviceOrderEvents and serviceOrderEventStatus are provided (${language})`, (): void =>
      expectComponentToMatchSnapshotForServiceOrderEvents(language));
  });
});
