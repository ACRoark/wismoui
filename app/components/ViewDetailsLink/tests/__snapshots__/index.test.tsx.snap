// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewDetailsLink /> should render the link when a valid serviceOrderEvents and serviceOrderEventStatus are provided (en) 1`] = `
<button
  className="dte-wismo-button dte-wismo-button-link"
  data-track="mimo"
  data-track-action="click"
  data-track-detail="order tracker"
  onClick={[Function]}
>
  View Details
</button>
`;

exports[`<ViewDetailsLink /> should render the link when a valid serviceOrderEvents and serviceOrderEventStatus are provided (es) 1`] = `
<button
  className="dte-wismo-button dte-wismo-button-link"
  data-track="mimo"
  data-track-action="click"
  data-track-detail="order tracker"
  onClick={[Function]}
>
  Ver Detalles
</button>
`;

exports[`<ViewDetailsLink /> should render the link when a valid verificationCategory, verificationCode, and wantDate are provided (en) 1`] = `
<button
  className="dte-wismo-button dte-wismo-button-link"
  data-track="mimo"
  data-track-action="click"
  data-track-detail="order tracker"
  onClick={[Function]}
>
  View Details
</button>
`;

exports[`<ViewDetailsLink /> should render the link when a valid verificationCategory, verificationCode, and wantDate are provided (es) 1`] = `
<button
  className="dte-wismo-button dte-wismo-button-link"
  data-track="mimo"
  data-track-action="click"
  data-track-detail="order tracker"
  onClick={[Function]}
>
  Ver Detalles
</button>
`;
